### ISSUE TITLE
Unable to add migrations with "dotnet-ef" - "Could not load file or assembly 'Remotion.Linq'"

### ISSUE THREAD

**Issue by joshcomley** [dotnet/efcore#4865]
**TL;DR**

Adding migrations results in: `Could not load file or assembly 'Remotion.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=fee00910d6e5f53b' or one of its dependencies. The system cannot find the file specified.`

Here is how to reproduce:

**project.json**

```
{
  "version": "1.0.0-*",
  "description": "Class Library",
  "authors": [ "Josh Comley" ],
  "tags": [ "" ],
  "projectUrl": "",
  "licenseUrl": "",
  "dependencies": {
    "Microsoft.EntityFrameworkCore": "1.0.0-*",
    "Microsoft.NETCore.Platforms": "1.0.1-*"
  },
  "tools": {
    "dotnet-ef": "1.0.0-*"
  },
  "frameworks": {
    "dnx451": { },
    "netstandardapp1.5": {
      "dependencies": {
        "NETStandard.Library": "1.0.0-*"
      },
      "imports": [
        "dnxcore50",
        "portable-net451+win8"
      ]
    }
  }
}
```

**class1.cs**

```
using Microsoft.EntityFrameworkCore;

namespace ClassLibrary1
{
    // This project can output the Class library as a NuGet Package.
    // To enable this option, right-click on the project and select the Properties menu item. In the Build tab select "Produce outputs on build".
    public class ApplicationDbContext : DbContext
    {
    }
}
```

**Then run**

`dotnet ef migrations add AnyOldMigration`

**Result**

```
D:\Code\TestApp\src\ClassLibrary1>dotnet ef migrations add AnyOldMigration
System.IO.FileNotFoundException: Could not load file or assembly 'Remotion.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=fee00910d6e5f53b' or one of its dependencies. The system cannot find the file specified.
File name: 'Remotion.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=fee00910d6e5f53b'
   at Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServiceCollectionExtensions.AddQuery(IServiceCollection serviceCollection)
   at Microsoft.EntityFrameworkCore.Infrastructure.EntityFrameworkServiceCollectionExtensions.AddEntityFramework(IServiceCollection serviceCollection)
   at Microsoft.EntityFrameworkCore.Internal.ServiceProviderCache.<>c__DisplayClass4_0.<GetOrAdd>b__1(Int64 k)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.EntityFrameworkCore.DbContext..ctor(DbContextOptions options)
   at ClassLibrary1.ApplicationDbContext..ctor()
Could not load file or assembly 'Remotion.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=fee00910d6e5f53b' or one of its dependencies. The system cannot find the file specified.
```

**Extra info**

I am assuming that this has something to do with the build being broken in general for dotnet CLR stuff, and a `dotnet restore` fails where Visual Studio's restore when saving `project.json` succeeds (as does `dnu restore`).

In the output of `dotnet restore` is:

```
Restoring Tool 'dotnet-ef' for 'D:\Code\TestApp\src\ClassLibrary1\obj\35856cde-3109-426a-99fa-5b6175ab66e7\bin\project.json' in 'D:\Code\TestApp\src\ClassLibrary1\obj\35856cde-3109-426a-99fa-5b6175ab66e7\bin'
info : Restoring packages for D:\Code\TestApp\src\ClassLibrary1\obj\35856cde-3109-426a-99fa-5b6175ab66e7\bin\project.json...
error: Ix-Async 1.2.5 is not compatible with DNXCore,Version=v5.0.
error: Remotion.Linq 2.0.2 is not compatible with DNXCore,Version=v5.0.
error: Some packages are not compatible with DNXCore,Version=v5.0.
error: Ix-Async 1.2.5 is not compatible with DNXCore,Version=v5.0 (win7-x64).
error: Remotion.Linq 2.0.2 is not compatible with DNXCore,Version=v5.0 (win7-x64).
error: Some packages are not compatible with DNXCore,Version=v5.0 (win7-x64).
error: Ix-Async 1.2.5 is not compatible with DNXCore,Version=v5.0 (win7-x86).
error: Remotion.Linq 2.0.2 is not compatible with DNXCore,Version=v5.0 (win7-x86).
error: Some packages are not compatible with DNXCore,Version=v5.0 (win7-x86).
info : Committing restore...
info : Restore failed in 5461ms.
Writing 'D:\Code\TestApp\src\ClassLibrary1\obj\35856cde-3109-426a-99fa-5b6175ab66e7\bin\project.lock.json' to 'D:\Code\TestApp\src\ClassLibrary1\artifacts\Tools\dotnet-ef\project.lock.json'
```

I figure once the build is restored, I can run `dotnet restore` and all will be alleviated, but just in case I thought it was worth reporting the problem.

**Comment by bricelam** [comment]
Possibly related to dotnet/cli#1181

**Comment by joshcomley** [comment]
Hi, thanks, I checked that out, and tried amending my framework imports accordingly but no luck.

I am completely unable to run any migrations code on the latest 20221. Do you have a working `dotnet ef` project?

**Comment by bricelam** [comment]
> Do you have a working `dotnet ef` project?

I get different errors every day. :smile: There is A LOT of churn in the CLI. I'm keeping #3925 open until things stabilize.

**Comment by joshcomley** [comment]
Ha, yeah I know it's a pain :D frustratingly I had a version of `dotnet ef` working about a week ago; I could add migrations etc. But now it's stopped working and I wish I could somehow point to the old stuff, but I can't figure it out.

**Comment by rowanmiller** [comment]
Closing as we are tracking all this via #3925
